mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -551,12 +551,16 @@ private[spark] class MesosClusterScheduler(
       currentOffers: List[ResourceOffer],
       tasks: mutable.HashMap[OfferID, ArrayBuffer[TaskInfo]]): Unit = {
     for (submission <- candidates) {
+      val acceptedResourceRoles = getAcceptedResourceRoles(submission.schedulerProperties)
       val driverCpu = submission.cores
       val driverMem = submission.mem
       logTrace(s"Finding offer to launch driver with cpu: $driverCpu, mem: $driverMem")
       val offerOption = currentOffers.find { o =>
-        getResource(o.resources, "cpus") >= driverCpu &&
-        getResource(o.resources, "mem") >= driverMem
+        val acceptableResources = o.resources.asScala
+          .filter((r: Resource) => acceptedResourceRoles(r.getRole))
```

Comment:
Just a style thing, but I think you'd generally just write `(r => acceptedResourceRoles(r.getRole))`